internal/client/client_batch.go

```diff
@@ -555,6 +555,8 @@ func (c *batchCommandsClient) waitConnReady() (err error) {
 			cancel()
 			break
 		}
+		// Trigger idle connection to reconnection
+		c.conn.Connect()
```

Comment:
This seems to be an experimental API which may not be stable enough?

Reply:
Besides, it seems each time the API is triggered a new goroutine would be created which could be a risk.

Reply:
Nice catch. For now, I've only found `Connect ` API to work, I try to use the TiKV RPC interface such as heartbeat or health check RPC API, but I didn't find such an interface.

Anyway, I just added a test about this.

Reply:
Actually `GetState` and `WaitForStateChange` are both experimental, I think it's ok to use `Connect` here. How about to trigger `Connect` only once (maybe just before the for loop) and only when the state is idle?

Reply:
`Connect ` API internal already checks the state, and only do connect when state is idle.

Reply:
`WaitForStateChange` has been used for some time and the risk is relatively low.
Maybe we'd better check the gRPC client source code in details about the state management and how could the `Connect` impact the underlying gRPC logic.